src/otx/algo/visual_prompting/zero_shot_segment_anything.py

```diff
@@ -0,0 +1,697 @@
+
+        # sample the highest score one of the samples that are in the same grid
+        points_scores = matched_grid[matched_grid[..., -1].argsort(dim=0, descending=True)[0]].diagonal().T
+
+        # sort by the highest score
+        points_scores = points_scores[torch.argsort(points_scores[:, -1], descending=True)]
+
+        return points_scores, bg_coords
+
+
+class ZeroShotSegmentAnything(SegmentAnything):
+    """Zero-shot learning module using Segment Anything."""
+
+    def __init__(
+        self,
+        default_threshold_reference: float = 0.3,
+        default_threshold_target: float = 0.65,
+        *args,
+        **kwargs,
+    ) -> None:
+        msg = ""
+        if len(kwargs) == 0:
+            msg += "There isn't any given argument. Default setting will be used."
+        elif len(kwargs) == 1 and "backbone" in kwargs:
+            msg += (
+                f"There is only backbone (={kwargs.get('backbone')}) argument. "
+                f"Other parameters will be set along with backbone (={kwargs.get('backbone')})."
+            )
+        elif "backbone" not in kwargs:
+            msg += "There isn't a backbone argument, it will be reset with backbone=tiny_vit."
+        if len(msg) > 0:
+            log.info(msg)
+            kwargs = self.set_default_config(**kwargs)
+
+        # check freeze conditions
+        for condition in ["freeze_image_encoder", "freeze_prompt_encoder", "freeze_mask_decoder"]:
+            if not kwargs.get(condition, False):
+                log.warning(f"{condition}(=False) must be set to True, changed.")
+                kwargs[condition] = True
+
+        super().__init__(*args, **kwargs)
+
+        self.prompt_getter = PromptGetter(image_size=self.image_size)
+        self.prompt_getter.set_default_thresholds(
+            default_threshold_reference=default_threshold_reference,
+            default_threshold_target=default_threshold_target,
+        )
+
+        self.point_labels_box = torch.tensor([[2, 3]], dtype=torch.float32)
+        self.has_mask_inputs = [torch.tensor([[0.0]]), torch.tensor([[1.0]])]
+
+        self.initialize_reference_info()
+
+    def set_default_config(self, **kwargs) -> dict[str, Any]:
+        """Set default config when using independently."""
+        backbone = kwargs.get("backbone", "tiny_vit")
+        kwargs.update(
+            {
+                "backbone": backbone,
+                "load_from": kwargs.get("load_from", DEFAULT_CONFIG_SEGMENT_ANYTHING[backbone]["load_from"]),
+                "freeze_image_encoder": kwargs.get("freeze_image_encoder", True),
+                "freeze_mask_decoder": kwargs.get("freeze_mask_decoder", True),
+                "freeze_prompt_encoder": kwargs.get("freeze_prompt_encoder", True),
+            },
+        )
+        return kwargs
+
+    def initialize_reference_info(self) -> None:
+        """Initialize reference information."""
+        self.reference_feats: Tensor = None
```

Comment:
I'm feeling that it is more natural to make it as [ParameterDict](https://pytorch.org/docs/stable/generated/torch.nn.ParameterDict.html#torch.nn.ParameterDict) with `requires_grad=False` because we update this values from the given data. And, I guess it makes also comfortable for handling its checkpoint.

Reply:
Done at https://github.com/openvinotoolkit/training_extensions/pull/2834/commits/121a03de8e1dbec252e2046e7de5ddb289ca18b8.